threat_model.md

```diff
@@ -0,0 +1,190 @@
+| Identities + repeatable data access                  | Full access to past and future data | No access                           |
+| App secret + repeatable data access                  | Full access to past data            | No access                           |
+| Identities modification + one-time data access       | Full access to past data            | No access                           |
+| Identities modification + repeatable data access     | Full access to past and future data | Full access to future data          |
+| Application Code control                             | Full access to past and future data | Full access to past and future data |
+
+### Detailed attacks analysis
+
+**One-time data access**:
+
+*Attack*: An attacker gains access to all user data stored by the *application*. This attack is not repeatable.   
+
+*Impact:* Without *Tanker*, the attacker immediately gains access to all user data existing at this point in time. With *Tanker*, they have no access to Tanker-encrypted data. 
+
+**Repeatable data access**:
+
+*Attack*: An attacker gains access to all user data stored by the *application*. This attack is repeatable. 
+
+*Impact:* Without *Tanker*, the attacker immediately gains access to all user data existing at this point in time. By repeating the same attack, they also have access to all future user data. With *Tanker*, they have no access to Tanker-encrypted data. 
+
+**Identities + repeatable data access**:
+
+*Attack*: An attacker gains access to all user data stored by the *application*. They also gain access to all stored *Tanker Identities*. This attack is repeatable. 
+
+*Impact:* Without *Tanker*, the attacker immediately gains access to all user data existing at this point in time. By repeating the same attack, they also have access to all future user data. With *Tanker*, they have no access to Tanker-encrypted data. 
+
+**App secret + repeatable data access**:
+
+*Attack*: An attacker gains access to all user data stored by the *application*. They also gain access to the private [Trustchain Signature Key Pair] contained in the app secret. This attack is repeatable. 
+
+*Impact:* Without *Tanker*, the attacker immediately gains access to all user data existing at this point in time. By repeating the same attack, they also have access to all future user data. With *Tanker*, they have no access to Tanker-encrypted data. 
+
+**Identity modification + one-time data access** :
+
+*Attack*: An attacker gains access to all user data stored by the *application*. They also gain access to all stored *Tanker Identities* and are able to change them. They replace all *user* *Identities* by their own. This attack is not repeatable. 
+
+*Impact:* Without *Tanker*, the attacker immediately gains access to all user data existing at this point in time. With *Tanker*, they have no access to Tanker-encrypted data. All future user data will however be encrypted for the attacker's *Tanker* account.
```

Comment:
How would an attacker manage to create devices for all users, say if email verification is the default verification method?

Wouldn't the attacker also need to control the app's code to force a specific verification method (e.g. background use of a passphrase or verificationKey?)

Reply:
no need. The attacker replaces every public identity by his one. He has access to encrypted data, so he takes it. New encrypted data will be encrypted for his public identity -> he can decrypt it normally